### PR TITLE
Add overview and resource info to README

### PR DESCRIPTION
**Pull request by mjordan56**
Added overview and Markdown syntax specification information with links to help users of the package understand Markdown syntax and how the package implements Markdown. Also added references to online Markdown viewer and additional resources.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,7 +5,61 @@
...
+## Overview
+
+The [flutter_markdown](https://pub.dev/packages/flutter_markdown) package
+utilizes the Dart [markdown](https://pub.dev/packages/markdown) package to
+create a Markdown widget comprised of Flutter widgets to render a rich text
+representation of plain text composed with a lightweight markup language.
+
+The markdown package parses Markdown syntax into HTML. The markdown package
+generates an Abstract Syntax Tree (AST) from the source Markdown data. The
+markdown package then uses the
```
Comment (domesticmouse) [inline README.md:19]: ditto

File: README.md [hunk 2]
```diff
@@ -5,7 +5,61 @@
...
+[Visitor pattern](https://en.wikipedia.org/wiki/Visitor_pattern) to traverse
+the AST nodes to create an HTML representation of the Markdown data. The Visitor
+pattern allows other visitor implementors to traverse the AST nodes and
+interpret them for other purposes such as printing out the tree or creating
+Flutter widgets.
+
+The flutter_markdown package uses the markdown package to parse the Markdown
```
Comment (domesticmouse) [inline README.md:26]: s/flutter_markdown package/`flutter_markdown` package/

File: README.md [hunk 3]
```diff
@@ -5,7 +5,61 @@
...
+flutter_markdown package doesn't support inline HTML.
+
+## Markdown Specifications and flutter_markdown Compliance
+
+There are three seminal documents regarding Markdown syntax; the
+[original Markdown syntax documentation](https://daringfireball.net/projects/markdown/syntax)
+specified by John Gruber, the
+[CommonMark specification](https://spec.commonmark.org/0.29/), and the
+[GitHub Flavored Markdown specification](https://github.github.com/gfm/).
+The CommonMark specification brings order to and clarifies the Markdown syntax
```
Comment (domesticmouse) [inline README.md:54]: add paragraph break here?

File: README.md [hunk 4]
```diff
@@ -5,7 +5,61 @@
...
+cases left ambiguous or unclear in the Gruber document. GitHub Flavored
+Markdown (GFM) is a strict superset of CommonMark used by GitHub.
+
+The markdown package, and in extension, the flutter_markdown package, supports
```
Comment (domesticmouse) [inline README.md:58]: `markdown` and `flutter_markdown`

**Review by domesticmouse — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -5,7 +5,61 @@
...
+HTML.
+
+## Overview
+
+The [flutter_markdown](https://pub.dev/packages/flutter_markdown) package
+utilizes the Dart [markdown](https://pub.dev/packages/markdown) package to
+create a Markdown widget comprised of Flutter widgets to render a rich text
+representation of plain text composed with a lightweight markup language.
+
+The markdown package parses Markdown syntax into HTML. The markdown package
```
[the remark this answers is not in the document]
Comment (mjordan56) [reply README.md]: Good suggestion. I'll perform a global search and update the relevant markdown and flutter_markdown package references.

File: README.md [hunk 6]
```diff
@@ -5,7 +5,62 @@
...
+Flutter widgets.
+
+The `flutter_markdown` package uses the `markdown` package to parse the Markdown
+source to generate an AST of element and text nodes. The AST element nodes
+have HTML element tags like h1, h2, p, code, etc. that map to Markdown syntax.
+The elements are either block or inline elements. The `flutter_markdown`
+MarkdownBuilder class implements the NodeVisitor interface from the `markdown`
+package to traverse the AST and build a tree of appropriate Flutter widgets.
+
+## Flutter Isn't HTML
```
Comment (domesticmouse) [inline README.md:33]: Suggestion: "Flutter isn't a web browser" 
Alternative: "Flutter isn't a HTML renderer"

thoughts?

File: README.md (hunk 1 above)
Comment (mjordan56) [reply README.md]: ditto as well for the fix.

File: README.md (hunk 3 above)
Comment (mjordan56) [reply README.md:55]: sure, paragraph break added.

File: README.md (hunk 4 above)
Comment (mjordan56) [reply README.md]: ditto on the change

**Review by domesticmouse — COMMENTED** [review]
I'm going to rope in @mit-mit to get a second pair of eyes on this change

File: README.md (hunk 6 above)
Comment (mjordan56) [reply README.md:33]: I like your alternative suggestion best. What I want to convey is that Flutter isn't interpreting and rendering HTML as a browser would do with HTML that is passed through. I'll make the change to your alternative suggestion.

**Comment by mjordan56** [comment]
@domesticmouse I think I may have screwed up that last re-review request. I thought the HTML renderer title change was still pending for review. I agree with having more eyes on the README.md changes. It never hurts to have another set of eyes review the changes. Thanks.

File: README.md (hunk 6 above)
Comment (domesticmouse) [reply README.md]: sounds good =)

**Review by domesticmouse — APPROVED** [review]
I'll wait for @mit-mit's approval before merging.

File: README.md [hunk 7]
```diff
@@ -5,7 +5,62 @@
...
+HTML.
+
+## Overview
+
+The [`flutter_markdown`](https://pub.dev/packages/flutter_markdown) package
+utilizes the Dart [`markdown`](https://pub.dev/packages/markdown) package to
+create a Markdown widget comprised of Flutter widgets to render a rich text
+representation of plain text composed with a lightweight markup language.
+
+The `markdown` package parses Markdown syntax into HTML. The `markdown` package
```
Comment (mit-mit) [inline README.md:17]: This paragraphs seems a bit too detailed; more consumers of this package are not likely to need details such as markdown using the visitor pattern. I'd do something simpler like:

`flutter_markdown` is built on top of the Dart [`markdown`](https://pub.dev/packages/markdown) package, which is uses to parse the markdown into an abstract syntax tree (AST), which represent an HTML representation of the Markdown data.

File: README.md [hunk 8]
```diff
@@ -5,7 +5,62 @@
 
 A markdown renderer for Flutter. It supports the
 [original format](https://daringfireball.net/projects/markdown/), but no inline
-html.
+HTML.
+
+## Overview
+
+The [`flutter_markdown`](https://pub.dev/packages/flutter_markdown) package
```
Comment (mit-mit) [inline README.md:12]: Remove the Dart part from here, as that is covered in the next paragraph, i.e. simplify to:

The [`flutter_markdown`](https://pub.dev/packages/flutter_markdown) package
can be used to render Markdown (a lightweight markup language) into a Flutter widget
containing a rich text representation.

File: README.md (hunk 7 above)
Comment (mjordan56) [reply README.md]: @mit-mit Thank you for your review comments. I've basically made the revisions you suggest in that I took the essence of your revision but I ran it through Grammarly.

File: README.md (hunk 8 above)
Comment (mjordan56) [reply README.md:12]: ditto. Suggested revisions made and ran through Grammarly.

**Review by mit-mit — APPROVED** [review]

**Comment by mit-mit** [comment]
Thanks for the contribution @mjordan56 !

**Comment by domesticmouse** [comment]
Thanks @mjordan56 for the word-smithing and @mit-mit for the review =)

**Pull request merged**